xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/htmlheader.vm

```diff
@@ -18,10 +18,11 @@
   #xwikiDataAttribute('wiki', $doc.wiki)##deprecated, use 'reference' instead
   #xwikiDataAttribute('space', $doc.space)##deprecated, use 'reference' instead
   #xwikiDataAttribute('page', $doc.name)##deprecated, use 'reference' instead
-  #xwikiDataAttribute('version', $doc.version)
-  #xwikiDataAttribute('rest-url', $services.rest.url($doc.documentReference))
+  #xwikiDataAttribute('version', $tdoc.version)
+  #xwikiDataAttribute('rest-url', $services.rest.url($tdoc.documentReferenceWithLocale))
   #xwikiDataAttribute('form-token', "$!{services.csrf.token}")
-  #xwikiDataAttribute('user-id', $!services.model.serialize($xcontext.userReference, 'default'))>
+  #xwikiDataAttribute('user-id', $!services.model.serialize($xcontext.userReference, 'default'))
+  #xwikiDataAttribute('locale', $tdoc.realLocale)>
```

Comment:
IMO you should use $tdoc.locale instead, otherwise it's very missleading. What you need most of the time technically is the locale of the document, if you feel it's really necessary you could add the locale of the content in a different metadata.

Reply:
For my knowledge what the difference between the locale of the document and the locale of the document's content?

Reply:
The default document locale is always "" (Locale.ROOT as in the last locale you get when you go up in the parents of the other locales). You can use the "default locale" property to indicate if the content of this document has been written in a specific locale ("en" for example for translation document usually) but this is just an information, it has no meaning at API level.

Reply:
From the `getRealLocale` method of `com.xpn.xwiki.api.Document`:
> Gets the real locale of the document. The real locale is either the default locale field when the locale field is empty (when the document is the default document) or the locale field otherwise when the document is a translation document

I had some issues when using `$tdoc.locale` as I was getting an empty string when the document was in english (default). OTOH `$doc.realLocale` is used in `xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/menus_language.vm` so I guess it's ok to use it.

Reply:
> I had some issues when using $tdoc.locale as I was getting an empty string when the document was in english (default)

You are mixing things actually, english is not always the default and you can see the previous comment for details on the rest. Empty string IS the locale of the document from API point of view so if there is only one indicate it should be this one, the other one is an information you may or may not indicate in the UI.

Reply:
ok so using `locale` instead of `realLocale` allows knowing if the content is written in the default locale or not, right? (if we were using `realLocale` we wouldn't have this info)

Reply:
> Empty string IS the locale of the document from API point of view

Ok I'll do the requested change then :)